.github/workflows/smoke-test.yml

```diff
@@ -48,25 +48,29 @@ jobs:
         # fix formatting with: cargo fmt --manifest-path=rust/Cargo.toml --all
         run: cargo fmt --manifest-path=Cargo.toml --all --check
 
-      - name: Build
-        run: cargo build --verbose --features untrusted,bindings,polars
-      
       - name: Check --no-default-features
-        run: cargo check --verbose --no-default-features --features untrusted,ffi,polars
-
-      - name: Test
-        run: cargo test --verbose --features untrusted,ffi,polars
+        run: cargo check --verbose --no-default-features --features untrusted,bindings
 
-      - name: Check that generated python is up to date
+      - name: Check that generated code is up to date
         run: git diff --exit-code
 
-      - name: Upload libs
+      - name: Build
+        run: cargo build --verbose --features untrusted,ffi,polars
+      
+      - name: Upload dynamic lib
         uses: actions/upload-artifact@v4
         with:
-          name: libs
-          path: |
-            rust/target/debug/libopendp.so
-            rust/target/debug/libopendp.a
+          name: dynamic-lib
+          path: rust/target/debug/libopendp.so
+
+      - name: Build (without Polars)
+        run: cargo build --verbose --features untrusted,ffi
+      
+      - name: Upload static lib
```

Comment:
```suggestion
      - name: Build (static w/o Polars)
        run: cargo build --verbose --features untrusted,ffi
      
      - name: Upload (static w/o Polars)
```

Reply:
Not sure how fast this is, but would it make any sense to split the two builds into separate jobs so they didn't need to be done sequentially? Maybe save that for down the road.

Reply:
It's actually super quick- about 12 seconds. The slow part is the testing, mainly since each doctest is compiled/linked separately.